### DISCUSSION TITLE
How come value changes are evaluated different with same results

### DISCUSSION THREAD

**Discussion by iamface** [community/community#22969] [New to GitHub]
<p>Hi everyone!</p>
<p>Can someone explain why these two changes are evaluated as different types of changes? The ending results are clearly the same, the values were changed from <code>string</code> to <code>int</code>, but the top change shows the entire value <strong>‘50’</strong> was replaced with <strong>50</strong> , and the bottom change shows only the apostrophes were removed.</p>
<p>I’m guessing it has something to do with them having different datatypes? Like a tinyint compared to an int? I notice in this PR every value with 1 or 2 characters (50, 24, 4, etc) has the full value replaced, but every value with 3 characters (200, 400, 287, etc) shows only the apostrophes replaced.</p>
<p><img src="https://user-images.githubusercontent.com/4822242/181074082-eddb6278-7e59-4860-803d-aa7649adf866.png" alt="Screen Shot 2019-11-01 at 2.51.33 PM.png" data-base62-sha1="juK2KIosMr8zfmLxeqhwQ5G7cXV" width="190" height="468"></p>

**Comment by seveas** [comment, marked as answer]
<p>The diff machinery doesn’t care about datatypes. Everying is text to it :smiley:</p>
<p>This type of diff is called a ‘word diff’ and I guess the preimage needs to be 5 characters or longer for this part of the diff machinery to kick in. Though that’s just a guess, I have no idea about the actual internals.</p>

**Comment by iamface** [comment]
<p>Thanks for the response <a class="mention" href="/seveas">@seveas</a>. Makese sense it’s all just evaluated as text and not actual types ha </p>
<p>I assume there’s some efficiency logic once the diff string hits different lengths, but I’m curious exactly what it might be, or why (if simply for machine efficiency).</p>
